kibana/client.go

```diff
@@ -346,6 +346,19 @@ func (client *Client) readVersion() error {
 // IgnoreVersion was set when creating the client.
 func (client *Client) GetVersion() version.V { return client.Version }
 
+// KibanaIsServerless returns true if we're talking to a serverless instance.
+// Right now we don't have an API to tell us if we're running against serverless or not, so this actual implementation is something of a hack.
+// see https://github.com/elastic/kibana/pull/164850
+func (client *Client) KibanaIsServerless() (bool, error) {
+	ret, _, err := client.Connection.Request("GET", "/api/saved_objects/_find", nil, nil, nil)
```

Comment:
If Kibana changes this, which is entirely possible because saved objects are one of the major APIs, then this might give us the wrong result at some point. Hopefully we can catch that in the snapshot builds before release.